README.rst

```diff
@@ -29,13 +29,86 @@ To authenticate, and then get a list of resources you have access to::
     for resource in hs.getResourceList():
         print(resource)
 
-To connect to a development HydroShare server::
+To authenticate using OAuth2 authentication (using a user and password supplied by the user), and then get a list of
+resources you have access to::
+
+    from oauthlib.oauth2 import TokenExpiredError
+    from hs_restclient import HydroShare, HydroShareAuthOAuth2
+
+    # Get a client ID and client secret by registering a new application at:
+    # https://www.hydroshare.org/o/applications/
+    # Choose client type "Confidential" and authorization grant type "Resource owner password-based"
+    # Keep these secret!
+    client_id = 'MYCLIENTID'
+    client_secret = 'MYCLIENTSECRET'
+
+    auth = HydroShareAuthOAuth2(client_id, client_secret,
+                                username='myusername', password='mypassword')
+    hs = HydroShare(auth=auth)
+
+    try:
+        for resource in hs.getResourceList():
+            print(resource)
+    except TokenExpiredError as e:
+        hs = HydroShare(auth=auth)
+        for resource in hs.getResourceList():
+            print(resource)
+
+Note that currently the client does not handle token renewal, hence the need to catch TokenExpiredError.
+
+To authenticate using OAuth2 authentication (using an existing token), and then get a list of resources you have
+access to::
+
+    from oauthlib.oauth2 import TokenExpiredError
+    from hs_restclient import HydroShare, HydroShareAuthOAuth2
+
+    # Get a client ID and client secret by registering a new application at:
+    # https://www.hydroshare.org/o/applications/
+    # Choose client type "Confidential" and authorization grant type "Resource owner password-based"
```

Comment:
@danames @jsadler2 Is this the correct grant type for the Tethys application?  I'm not sure what you are using.